File: Labelizer-Backend/src/main/java/com/raphaelcoutu/labelizer/entity/LabelBox.java
package com.raphaelcoutu.labelizer.entity;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Data
@Table(name = "label_box")
public class LabelBox {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "photo_id", nullable = false)
    private Photo photo;

    @ManyToOne
    @JoinColumn(name = "label_id", nullable = false)
    private Label label;

    @NotNull
    @Column(name = "x")
    private Integer x;

    @NotNull
    @Column(name = "y")
    private Integer y;

    @NotNull
    @Column(name = "width")
    private Integer width;

    @NotNull
    @Column(name = "height")
    private Integer height;
}

File: Labelizer-Frontend/src/app/_models/label-box.ts
import { Label } from './label';

export class LabelBox {
  id?: number;
  x: number;
  y: number;
  width: number;
  height: number;
  label: Label;

  constructor(x: number, y: number, width: number, height: number, label: Label){
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.label = label;`d`
  }

  toString() {
    return 'x:' + this.x + ', y:' + this.y + ', w:' + this.width + 'h:' + this.height;
  }
}
